msmbuilder/decomposition/kernel_approx.py

```diff
@@ -0,0 +1,103 @@
+# Author: Carlos Xavier Hernandez <cxh@stanford.edu>
+# Contributors: Muneeb Sultan <msultan@stanford.edu>, Evan Feinberg <enf@stanford.edu>
+# Copyright (c) 2015, Stanford University and the Authors
+# All rights reserved.
+
+from __future__ import absolute_import
+
+import numpy as np
+from scipy.linalg import svd
+
+from sklearn import kernel_approximation
+from sklearn.metrics.pairwise import pairwise_kernels
+
+from .base import MultiSequenceDecompositionMixin
+from ..utils import check_iter_of_sequences
+
+
+__all__ = ['Nystroem', 'LandmarkNystroem']
+
+
+class Nystroem(MultiSequenceDecompositionMixin, kernel_approximation.Nystroem):
+    __doc__ = kernel_approximation.Nystroem.__doc__
+
+
+class LandmarkNystroem(Nystroem):
+    """Approximate a kernel map using a subset of the training data.
+    Constructs an approximate feature map for an arbitrary kernel
+    using a subset of the data as basis.
+    Read more in the :ref:`User Guide <nystroem_kernel_approx>`.
+    Parameters
+    ----------
+    landmarks : ndarray of shape (n_frames, n_features)
+        Custom landmark points for the Nyostroem approximation
+    kernel : string or callable, default="rbf"
+        Kernel map to be approximated. A callable should accept two arguments
+        and the keyword arguments passed to this object as kernel_params, and
+        should return a floating point number.
+    n_components : int
+        Number of features to construct.
+        How many data points will be used to construct the mapping.
+    gamma : float, default=None
+        Gamma parameter for the RBF, polynomial, exponential chi2 and
+        sigmoid kernels. Interpretation of the default value is left to
+        the kernel; see the documentation for sklearn.metrics.pairwise.
+        Ignored by other kernels.
+    degree : float, default=3
+        Degree of the polynomial kernel. Ignored by other kernels.
+    coef0 : float, default=1
+        Zero coefficient for polynomial and sigmoid kernels.
+        Ignored by other kernels.
+    kernel_params : mapping of string to any, optional
+        Additional parameters (keyword arguments) for kernel function passed
+        as callable object.
+    random_state : {int, RandomState}, optional
+        If int, random_state is the seed used by the random number generator;
+        if RandomState instance, random_state is the random number generator.
+    Attributes
+    ----------
+    components_ : array, shape (n_components, n_features)
+        Subset of training points used to construct the feature map.
+    component_indices_ : array, shape (n_components)
+        Indices of ``components_`` in the training set.
+    normalization_ : array, shape (n_components, n_components)
+        Normalization matrix needed for embedding.
+        Square root of the kernel matrix on ``components_``.
+    References
+    ----------
+    * Williams, C.K.I. and Seeger, M.
```

Comment:
You should also add it to the docs :) Should just be a matter of listing the relevant classes in decomposition.rst